enum.c

```diff
@@ -4552,12 +4559,25 @@ uniq_iter(RB_BLOCK_CALL_FUNC_ARGLIST(i, hash))
 
 /*
  *  call-seq:
- *     enum.uniq                -> new_ary
- *     enum.uniq { |item| ... } -> new_ary
+ *    uniq                  -> array
+ *    uniq {|element| ... } -> array
+ *
+ *  With no block, returns a new array containing only unique elements;
+ *  the array has no two elements +e0+ and +e1+ such that <tt>e0 == e1</tt>:
  *
- *  Returns a new array by removing duplicate values in +self+.
+ *    %w[a b c c b a a b c].uniq       # => ["a", "b", "c"]
+ *    [0, 1, 2, 2, 1, 0, 0, 1, 2].uniq # => [0, 1, 2]
+ *
+ *  With a block, returns a new array containing only for which the block
+ *  returns a unique value:
+ *
+ *    a = [0, 1, 2, 3, 4, 5, 5, 4, 3, 2, 1]
+ *    a.uniq {|i| i.even? ? i : 0 } # => [0, 2, 4]
+ *    a = %w[a b c d e e d c b a a b c d e]
+      a.uniq {|c| c < 'c' }         # => ["a", "c"]
+ *
+ *  Among equal elements, the elements retained is indeterminate.
```

Comment:
I'm not sure if it is spec, but I think all ruby implementations returns the first element with the unique value in such cases.  I don't think we need to mention the results are indeterminate.  We could say the results are implementation-defined, but unless we can find a counter case, I don't have a problem stating that the first unique element is retained.  The previous language implied that already, with `removing duplicate values`, since the first value would not be duplicate, but other equal values would be, and those are the ones that are removed.

Reply:
Removed.